src/Link.cc

```diff
@@ -213,8 +264,20 @@ uint64_t Link::VisualCount() const
 /////////////////////////////////////////////////
 const Visual *Link::VisualByIndex(const uint64_t _index) const
 {
-  if (_index < this->dataPtr->visuals.size())
-    return &this->dataPtr->visuals[_index];
+  return this->dataPtr->VisualByIndex(_index);
+}
+
+/////////////////////////////////////////////////
+Visual *Link::VisualByIndex(uint64_t _index)
+{
+  return const_cast<Visual*>(this->dataPtr->VisualByIndex(_index));
```

Comment:
This is fine, but we could avoid creating all the helper functions by keeping the implementation in the original const function (i.e `Link::VisualByIndex() const`) and calling that from here via `const_cast`: https://stackoverflow.com/a/123995/283225

Reply:
Nice, the double cast. Done in 9ef6f94. I'll make similar changes to the other PRs.